editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx

```diff
@@ -285,7 +324,62 @@ export const ValidationChecks = (props: {
           </ListItem>
         ))}
       </List>
-      <Divider />
+    </Box>
+  );
+};
+
+const ValidationSummary = (props: { validationChecks: ValidationCheck[] }) => {
+  const { validationChecks } = props;
+  const atLeastOneFail =
+    validationChecks.filter((check) => check.status === "Fail").length > 0;
+  const countByStatus = countBy(validationChecks, "status");
+
+  const summary: string[] = [];
+  Object.entries(countByStatus).map(([status, count]) => {
+    switch (status) {
+      case "Fail":
+        summary.push(`${count} failing`);
+        break;
+      case "Warn":
+        summary.push(count === 1 ? `${count} warning` : `${count} warnings`);
+        break;
+      case "Pass":
+        summary.push(`${count} successful`);
+        break;
+      case "Not applicable":
+        summary.push(`${count} skipped`);
+        break;
+    }
+  });
+  // If there aren't any fails, still add "0 errors" to end of summary string for distinction from "warnings"
+  const formattedSummary = atLeastOneFail
+    ? summary.join(", ")
+    : summary.join(", ").concat(", 0 errors");
+
+  return (
+    <Box sx={{ display: "flex", alignItems: "center" }}>
+      {atLeastOneFail ? (
+        <Cancel
+          color="error"
+          fontSize="large"
+          sx={{ minWidth: (theme) => theme.spacing(5.5) }}
+        />
+      ) : (
+        <CheckCircle
+          color="success"
+          fontSize="large"
+          sx={{ minWidth: (theme) => theme.spacing(5.5) }}
+        />
```

Comment:
A small question: I'm using icons with a fill color for the top-level summary, and generally icons without fill for the list items (again similar to Github CI UI) - with the exception of the warning icon, should we switch it to be consistent? Any suggestions if so?

Reply:
The icons look fine to me, I'd generally favour using the best icon for the job over a hardline rule on fill/border styles.

Reply:
Makes sense ! Just made a slight tweak to use the outlined version of this one - which because "warnings" were the most confusing point of feedback perhaps will help with making them less prominent / more obviously "not errors"
![Screenshot from 2024-07-30 13-59-59](https://github.com/user-attachments/assets/c79626f3-252d-4c3b-ad96-706fe2897e9b)